sys/arduino/Makefile.arduino_sketches.inc

```diff
@@ -0,0 +1,31 @@
+# Compile together the Arduino sketches of the application
+# They are declared a as new module $(SKETCH_MODULE) in $(SKETCH_MOD_DIR)
+
+ifndef SKETCH_MODULE
+  $(error SKETCH_MODULE undefined. It should be defined to the sketches module name)
+endif
+ifndef SKETCH_MODULE_DIR
+  $(error SKETCH_MODULE_DIR undefined. It should be defined to the sketches module directory)
+endif
+ifndef SKETCHES
+  $(error SKETCHES undefined. It should be defined to the list of sketches files)
+endif
+
+SNIPDIR         = $(RIOTBASE)/sys/arduino
+SKETCHES_ALL    = $(SNIPDIR)/pre.snip $(SKETCHES) $(SNIPDIR)/post.snip
+SKETCH_CPP     ?= arduino_sketches.cpp
+
+# Building the module files
+#   Do not use $^ in receipes as Makefile is also a prerequisite
+$(SKETCH_MODULE_DIR)/Makefile: $(SKETCH_MODULE_DIR)/$(SKETCH_CPP)
+	$(Q)echo 'SRCXX = $(SKETCH_CPP)'               > $@
+	$(Q)echo 'include $$(RIOTBASE)/Makefile.base' >> $@
+$(SKETCH_MODULE_DIR)/$(SKETCH_CPP): $(SKETCHES_ALL) | $(SKETCH_MODULE_DIR)
+	$(Q)cat $(SKETCHES_ALL) > $@
+$(SKETCH_MODULE_DIR):
+	@mkdir -p $@
+
+# Make everything rebuild if current makefile changes
+CURMK = $(word $(words $(MAKEFILE_LIST)),$(MAKEFILE_LIST))
```

Comment:
You could use the lastword function instead

Reply:
Better use a different name than CURMK. This file is included from the Makefile.include which means it will pollute the name space.

Reply:
I will change to `lastword` I learned it after writing this, and update to a namespaced variable name.